diff --git a/drivers/led_strip/ws2812_spi.c b/drivers/led_strip/ws2812_spi.c
index 6832e32ba6..4aa8ba0e12 100644
--- a/drivers/led_strip/ws2812_spi.c
+++ b/drivers/led_strip/ws2812_spi.c
@@ -124,7 +124,7 @@ static int ws2812_strip_update_rgb(const struct device *dev,
 			switch (cfg->color_mapping[j]) {
 			/* White channel is not supported by LED strip API. */
 			case LED_COLOR_ID_WHITE:
-				pixel = 0;
+				pixel = pixels[i].w;
 				break;
 			case LED_COLOR_ID_RED:
 				pixel = pixels[i].r;
diff --git a/include/drivers/led_strip.h b/include/drivers/led_strip.h
index d8636eb545..29b41bd387 100644
--- a/include/drivers/led_strip.h
+++ b/include/drivers/led_strip.h
@@ -49,6 +49,8 @@ struct led_rgb {
 	uint8_t g;
 	/** Blue channel */
 	uint8_t b;
+	/** White channel */
+	uint8_t w;
 };
 
 /**
